From df3f95cb4c29b1d2612478159c28e66fc1dd8a76 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 20 Aug 2024 09:47:39 +0300
Subject: [PATCH] [FSW-7256] fs1: declare rr context and access type

With this patch, declared a structure with
register fields which are fixed and required
to configure range registers for a particular
address space which requires access.

Change-Id: Icf343f19b52e0f3056033f4122aaa9f673564f36
---

diff --git a/arch/fs1/arc/dev_security.h b/arch/fs1/arc/dev_security.h
index 23bde1e..59d0203 100644
--- a/arch/fs1/arc/dev_security.h
+++ b/arch/fs1/arc/dev_security.h
@@ -50,6 +50,20 @@
 	LBW_RR_INVALID_IDX = 16,
 };
 
+enum security_access {
+	SEC_ACCESS_RD_ONLY,
+	SEC_ACCESS_WR_ONLY,
+	SEC_ACCESS_RD_WR,
+	SEC_ACCESS_INVALID,
+};
+
+struct security_rr_ctx {
+	uint8_t rr_idx;
+	uint8_t acc_type;
+	uint32_t data_min;
+	uint32_t data_max;
+};
+
 void security_load_revoke_bits(void);
 int security_efuse_image_validation_en(void);
 void boot_rom_eom_overrides(void);
